include/jinja2cpp/user_callable.h

```diff
@@ -15,6 +17,86 @@ struct CanBeCalled : std::false_type {};
 template<typename T>
 struct CanBeCalled<T, typename std::enable_if<std::is_same<typename T::result_type, Value>::value>::type> : std::true_type {};
 
+template<typename T>
+struct ArgPromoter
+{
+    ArgPromoter(const T* val) : m_ptr(val) {}
```

Comment:
maybe something like gsl::not_null?

Reply:
or at least https://en.cppreference.com/w/cpp/utility/functional/reference_wrapper ?

Reply:
See no reason for that here. It's internal utility class which isn't directly exposed to the user.